### PR TITLE
bpo-35714: Reject null characters in struct format strings

### PR DESCRIPTION
**Pull request by ZackerySpytz**
struct.error is now raised if there is a null character in a struct
format string.

https://bugs.python.org/issue35714

### CODE REVIEW

**Review by mdickinson — APPROVED** [review]
LGTM

File: Lib/test/test_struct.py [hunk 1]
```diff
@@ -652,6 +652,13 @@ def test_format_attr(self):
         s2 = struct.Struct(s.format.encode())
         self.assertEqual(s2.format, s.format)
 
+    def test_issue35714(self):
+        # Embedded null characters should not be allowed in format strings.
+        for s in '\0', '\144\u0064\000xf', 'd\0d', '>ih\0', '=Q\0\0':
```
Comment (serhiy-storchaka) [inline Lib/test/test_struct.py]: What is the difference between all these cases? Why they are needed?

Comment (ZackerySpytz) [reply Lib/test/test_struct.py]: The embedded null is located at different positions of the string.

Comment (serhiy-storchaka) [reply Lib/test/test_struct.py]: And why it is handled differently depending on the position?

I would test with `'\0'`, `'2\0i'` and `b'\0'`.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**